src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -2058,10 +2060,37 @@ protected void parseHttpPort(String webAppMessage, int portPrefixIndex) throws P
         }
     }
 
+    /**
+     * Extracts the "message" field value from a JSON string.
+     *
+     * @param jsonString The JSON string to parse
+     * @return The value of the "message" field if present, null otherwise
+     */
+    private static String getMessageKeyFromJson(String jsonString) {
+        if (jsonString == null || jsonString.isEmpty()) {
+            return null;
+        }
+        try {
+            JSONObject jsonObject = new JSONObject(jsonString);
+            return jsonObject.optString("message", null);
+        } catch (JSONException e) {
+            // If parsing fails, it's not a valid JSON object
+            return null;
+        }
+    }
+
     protected void parseHttpsPort(List<String> messages) throws PluginExecutionException {
         for (String message : messages) {
             debug("Looking for https port in message: " + message);
-            String httpsMessageContents = message.split(LISTENING_ON_PORT_MESSAGE_PREFIX)[1];
+            String httpsMessageContents;
+            // check message format. if message is a json, only extract "message" node
+            String messageFromJson = getMessageKeyFromJson(message);
+            if (messageFromJson != null) {
+                debug("Message is in json format");
+                httpsMessageContents = messageFromJson.split(LISTENING_ON_PORT_MESSAGE_PREFIX)[1];
+            } else {
+                httpsMessageContents = message.split(LISTENING_ON_PORT_MESSAGE_PREFIX)[1];
+            }
```

Comment:
Now that you are only looking at the "message" part of the message, why wouldn't you check for "-ssl" before splitting on spaces and iterating through the parts of that specific message?